crates/rpc/rpc/src/trace.rs

```diff
@@ -204,12 +206,30 @@ where
                     let config = tracing_config(&trace_types);
                     let mut inspector = TracingInspector::new(config);
                     let (res, _) = inspect(&mut db, env, &mut inspector)?;
-                    let trace_res = inspector.into_parity_builder().into_trace_results_with_state(
-                        res,
-                        &trace_types,
-                        &db,
-                    )?;
+                    let ResultAndState { result, state } = res;
+
+                    let mut trace_res =
+                        inspector.into_parity_builder().into_trace_results(result, &trace_types);
+
+                    // If statediffs were requested, populate them with the account balance and
+                    // nonce from pre-state
+                    if let Some(ref mut state_diff) = trace_res.state_diff {
+                        populate_account_balance_nonce_diffs(
+                            state_diff,
+                            &db,
+                            state.iter().map(|(addr, acc)| (*addr, acc.info.clone())),
+                        )?;
+                    }
+
                     results.push(trace_res);
+
+                    // need to apply the state changes of this call before executing the
+                    // next call
+                    if calls.peek().is_some() {
+                        // need to apply the state changes of this call before executing
+                        // the next call
+                        db.commit(state)
```

Comment:
this is only temporary right - won't actually write to disk

Reply:
yes, this is only the cached, in mem db